web/src/components/logs-metrics.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Chart,
+  ChartAxis,
+  ChartGroup,
+  ChartLegendTooltip,
+  ChartLine,
+  ChartThemeColor,
+  createContainer,
+  getResizeObserver,
+} from '@patternfly/react-charts';
+import { Alert } from '@patternfly/react-core';
+import React from 'react';
+import { DateFormat, dateToFormat, getTimeFormatFromTimeRange } from '../date-utils';
+import { isMatrixResult, QueryRangeResponse, TimeRange } from '../logs.types';
+import { TestIds } from '../test-ids';
+import { intervalFromTimeRange, numericTimeRange } from '../time-range';
+import { CenteredContainer } from './centered-container';
+
+type MetricsData = { name: string; data: Array<{ name: string; x: number; y: number }> };
+type Domain = [number, number];
+interface LogsMetricsProps {
+  logsData?: QueryRangeResponse;
+  timeRange?: TimeRange;
+  isLoading?: boolean;
+  error?: unknown;
+}
+
+const GRAPH_HEIGHT = 250;
+
+const matrixToMetricsData = (
+  response?: QueryRangeResponse,
+): { data: Array<MetricsData> | undefined; xDomain: Domain; yDomain: Domain } => {
+  if (!response) {
+    return { data: undefined, xDomain: [0, 1], yDomain: [0, 1] };
+  }
+
+  if (!isMatrixResult(response.data)) {
+    return { data: undefined, xDomain: [0, 1], yDomain: [0, 1] };
+  }
+
+  const values = response.data.result;
+  const xDomain: Domain = [Number.MAX_VALUE, Number.MIN_VALUE];
+  const yDomain: Domain = [Number.MAX_VALUE, Number.MIN_VALUE];
+
+  const data = values.map((value) => {
+    const seriesName = JSON.stringify(value.metric);
+
+    return {
+      name: seriesName,
+      data: value.values.map((coordinate) => {
+        const time = parseInt(String(coordinate[0])) * 1000;
+
+        if (time < xDomain[0]) {
+          xDomain[0] = time;
+        }
+        if (time > xDomain[1]) {
+          xDomain[1] = time;
+        }
+
+        const y = parseFloat(String(coordinate[1]));
+
+        if (y < yDomain[0]) {
+          yDomain[0] = y;
+        }
+        if (y > yDomain[1]) {
+          yDomain[1] = y;
+        }
+
+        return { x: time, y, name: seriesName };
+      }),
+    };
+  });
+
+  return { data, xDomain, yDomain };
+};
+
+const CursorVoronoiContainer = createContainer('voronoi', 'cursor');
+
+export const LogsMetrics: React.FC<LogsMetricsProps> = ({
+  logsData,
+  isLoading,
+  error,
+  timeRange,
+}) => {
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+  const [width, setWidth] = React.useState<number>(0);
```

Comment:
Console has a hook for handling resize: https://github.com/openshift/console/blob/master/frontend/public/components/utils/ref-width-hook.ts, which has been in use for a while so the code should be well tested.

Ideally we would expose it to the SDK, but I don't think that should hold up this PR.